### PR TITLE
Anagram groups

### PR DESCRIPTION
**Pull request by lensman75**
Can't finish. Lost somewhere in the end. Some logic missed. Help me please.

### CODE REVIEW

File: anagramGroups.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
+let strs = ['eat', 'tea', 'tan', 'ate', 'nat', 'bat'];
+
+let group = new Array();
```
Comment (mkl-) [inline anagramGroups.js]: Why not simply `let group = []` ?

File: anagramGroups.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+function compareObjects(item1, item2) {
```
Comment (mkl-) [inline anagramGroups.js:5]: `compareObjects` are incorrect
image

File: anagramGroups.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  let obj1 = {};
+  let obj2 = {};
+  for (let i = 0; i < item1.length; i++) {
+    if (!(item1[i] in obj1)) {
+      obj1[item1[i]] = 1;
+    } else {
+      obj1[(item1[i] = obj1[item[i]] + 1)];
```
Comment (mkl-) [inline anagramGroups.js:12]: What is this?
`obj1` is not modified?

File: anagramGroups.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    }
+  }
+  //   obj1;
+  for (let i = 0; i < item2.length; i++) {
+    if (!(item2[i] in obj2)) {
+      obj2[item2[i]] = 1;
+    } else {
+      obj2[(item2[i] = obj2[item[i]] + 1)];
```
Comment (mkl-) [inline anagramGroups.js:19]: What is this?
`obj2` is not modified?

File: anagramGroups.js (hunk 2 above)
Comment (mkl-) [reply anagramGroups.js:5]: This function is over complicated. I cannot get an idea behind it. Could you please explain?

File: learnjsObjects.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
+// // https://learn.javascript.ru/object
```
Comment (mkl-) [inline learnjsObjects.js]: What is the purpose of this file?

File: permutableString.js [hunk 6]
```diff
@@ -0,0 +1,47 @@
+// https://dmitripavlutin.com/how-to-compare-objects-in-javascript/
+//Link above - compare objects, 3 subheader - don't clear 'Object.keys'...
+//Works when both of strings are the same length.
+function permutable(s1, s2) {
+    let obj1 = {}, obj2 = {};
+    function countSymbols(str) {
```
Comment (mkl-) [inline permutableString.js:6]: Please move this function outside.

File: permutableString.js [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+            } else {
+                obj[str[i]] += 1;
+            }
+        }
+        return obj;
+    }
+    obj1 = countSymbols(s1);
+    obj2 = countSymbols(s2);
+
+    countSymbols(s1, obj1);
```
Comment (mkl-) [inline permutableString.js:20]: Please delete this. Symbols are already counted in `obj1 = countSymbols(s1);`

File: permutableString.js [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+    countSymbols(s2, obj2);
+    const keys1 = Object.keys(obj1);
```
Comment (mkl-) [inline permutableString.js:22]: I guess there should be a call to `compareObjects(obj1, obj2)` ?

**Review by mkl- — CHANGES_REQUESTED** [review]

File: anagramGroups.js (hunk 1 above)
Comment (lensman75) [reply anagramGroups.js]: Agree! Probably it's due to my still low practice level. Fixed.

File: anagramGroups.js (hunk 2 above)
Comment (lensman75) [reply anagramGroups.js:5]: Yes, I found mistakes and fixed. Thanks.

**Pull request closed without merging**